src/Apps/Authentication/Hooks/useAuthDialogOptions.tsx

```diff
@@ -0,0 +1,107 @@
+import { AuthContextModule, AuthIntent } from "@artsy/cohesion"
+import { useTitleWithIntent } from "Apps/Authentication/Hooks/useTitleWithIntent"
+import { DEFAULT_TITLES } from "Components/AuthDialog/AuthDialog"
+import {
+  AuthDialogMode,
+  DEFAULT_AUTH_MODAL_INTENTS,
+  useAuthDialogContext,
+} from "Components/AuthDialog/AuthDialogContext"
+import { useEffect } from "react"
+import { useRouter } from "System/Router/useRouter"
+import { AfterAuthAction } from "Utils/Hooks/useAuthIntent"
+
+/**
+ * Converts query params into auth dialog options and sets them.
+ * Returns strings for use on parent page.
+ *
+ * **Valid query params:**
+ * - `title`: Sets the title of the screen
+ * - `description`: Sets the description
+ * - `intent`: If present for analytics will also set a default title if `title` isn't present
+ * - `afterAuthAction`: Sets `options.afterAuthAction`
+ * - `redirectTo`: Sets `options.redirectTo`
+ * - `contextModule`: Sets `analytics.contextModule`
+ *
+ * **Deprecated query params:**
+ * - `copy`: Sets the title of the screen
+ * - `afterSignUpAction`: Sets `options.afterAuthAction`
+ * - `submissionId`: If present will construct an `afterAuthAction` from it
+ */
```

Comment:
So rather than configure the auth dialog via a function like elsewhere on the site, when we're rendering it into the page, we configure it via query params. I've documented the supported params and the deprecated ones as well.

Reply:
nice